File: lib/JFFNetwork/Categories/NSURL+Cookies.m
#import "NSURL+Cookies.h"

@implementation NSURL (Cookies)

- (void)logCookies
{
    NSMutableString *cookiesLog = [[NSMutableString alloc] initWithFormat:@"Cookies for url: %@\n", self];
    
    NSArray *cookies = [[NSHTTPCookieStorage sharedHTTPCookieStorage] cookiesForURL:self];
    for (NSHTTPCookie *cookie in cookies) {
        
        [cookiesLog appendFormat:@"Name: '%@'; Value: '%@'\n", cookie.name, cookie.value];
    }
    
    NSLog(@"%@", cookiesLog);
}

- (void)removeCookies
{
    NSArray *cookies = [[NSHTTPCookieStorage sharedHTTPCookieStorage] cookiesForURL:self];
    for (NSHTTPCookie *cookie in cookies) {
        
        [[NSHTTPCookieStorage sharedHTTPCookieStorage] deleteCookie:cookie];
    }
}

@end

File: iAsync.Swift/lib/JNetwork/JNetwork/Extensions/NSURL+Cookies.swift
//
//  NSURL+Cookies.swift
//  JNetwork
//
//  Created by Vladimir Gorbenko on 24.09.14.
//  Copyright (c) 2014 EmbeddedSources. All rights reserved.
//

import Foundation

public extension NSURL {

    func logCookies() {
        
        var cookiesLog = "Cookies for url: \(self)\n"
    
        if let cookies = NSHTTPCookieStorage.sharedHTTPCookieStorage().cookiesForURL(self) as? [NSHTTPCookie] {
            
            for cookie in cookies {
                
                cookiesLog += "Name: '\(cookie.name)'; Value: '\(cookie.value)'\n"
            }
        }
    
        NSLog(cookiesLog)
    }
    
    func removeCookies() {
        
        let cookieStorage = NSHTTPCookieStorage.sharedHTTPCookieStorage()
        let cookies = cookieStorage.cookiesForURL(self) as? [NSHTTPCookie]
        
        if let cookies = cookies {
            for cookie in cookies {
            
                cookieStorage.deleteCookie(cookie as NSHTTPCookie)
            }
        }
    }
}
